ksql-engine/src/test/java/io/confluent/ksql/ddl/commands/CommandFactoriesTest.java

```diff
@@ -78,14 +79,31 @@ public void shouldCreateCommandForCreateStream() {
 
   @Test
   public void shouldCreateCommandForCreateTable() {
+    HashMap<String, Expression> tableProperties = new HashMap<>();
+    tableProperties.putAll(properties);
+    tableProperties.put(DdlConfig.KEY_NAME_PROPERTY, new StringLiteral("col1"));
     final DDLCommand result = commandFactories.create(
         new CreateTable(QualifiedName.of("foo"),
-            Collections.emptyList(), true, properties),
+            Collections.emptyList(), true, tableProperties),
         Collections.emptyMap());
 
     assertThat(result, instanceOf(CreateTableCommand.class));
   }
 
+  @Test
+  public void shouldFailCreateTableIfKeyIsNotProvided() {
```

Comment:
Shouldn't we add tests which check the expected results when the explicitly specified key field doesn't exist is not the same as the key in the message.